### DISCUSSION TITLE
Provide an easy way to disable multiple workflows

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#14485] [Actions]
As hinted at in #14480, it should be easy to disable a whole bunch of workflows at once.

The current process is actually quite painful for various reasons:
1. Not all views allow one to disable a workflow (ok, you can probably only disable it from one very specific view).
   * You can't disable a workflow from a [run's summary](https://github.com/check-spelling/superset/actions/runs/2109569185) even if it has been running for ¾⌛ 
      <img width="1388" alt="image" src="https://user-images.githubusercontent.com/2119212/162232455-8d8d642a-5ae0-4f7f-989d-067c5b050d47.png">
   * You can't disable a workflow from a [job](https://github.com/check-spelling/superset/runs/5870247055?check_suite_focus=true) even if it has been running for ¾⌛ 
      <img width="1400" alt="image" src="https://user-images.githubusercontent.com/2119212/162232675-d9ce526b-7640-44b8-8403-3b44dc58940e.png">
   * You can't disable a workflow from the [list of workflow runs](https://github.com/check-spelling/superset/actions)
      <img width="1106" alt="image" src="https://user-images.githubusercontent.com/2119212/162232957-09ef78a1-8f5e-4440-a334-43eecbdb380a.png">
2. When you're at the [right view](https://github.com/check-spelling/superset/actions/workflows/misc.yml)
   * you have to click:
      ![image](https://user-images.githubusercontent.com/2119212/162233475-06d0f583-392f-4e6e-9f68-6a958437edc6.png)
   * and click:
      ![image](https://user-images.githubusercontent.com/2119212/162233670-737845b0-bc3e-4873-8796-0dfdae098154.png)
   * and **wait**! -- if you don't wait, it's possible that if you click to go to shoot the next workflow, your disable request might be cancelled!
   * And then you have to go click on another workflow to start the process over again.

Anyway, this specific project only had 9 workflows I needed to disable, some have dozens.

Windows Explorer has a mode where there can be checkboxes to the side of objects that allow one to multi-select and do things:
![image](https://user-images.githubusercontent.com/2119212/162234304-d4b0e527-fe7c-40cc-8b0b-ff7f7c9ce592.png)

Yes, advanced users can try to use `control`+click, or `shift`+click, or mouse-drag to create range selections. These are all nifty things. But most of these metaphors don't work well in web pages. And they're very a11y hostile. Checkboxes are easy for everyone to use. GitHub supports [task lists](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists), so it definitely has some support for checkboxes.

Issues and Pull requests even support using check boxes to make mass changes:
![image](https://user-images.githubusercontent.com/2119212/162237871-09044f40-95b9-429d-9e6b-c361dabb4f96.png)

![image](https://user-images.githubusercontent.com/2119212/162238186-65193b58-9511-48da-9bf8-e45e047ee19d.png)

---

In case people are wondering, I'm using a MacBook Pro, and have been for the past 5 years. I'm not wed to a given platform, but when someone gets something right, I will recommend copying their UX to others who haven't.

**Comment by silverqx** [comment]
I'm missing this feature so much, mass enable/disable workflows and also mass delete workflow runs. Life would be much easier with them.

**Reply by jsoref** [reply]
Fwiw, you can now use the cli to disable workflows (one at a time https://cli.github.com/manual/gh_workflow_disable) or delete runs (one at a time https://cli.github.com/manual/gh_run_delete).

**Reply by silverqx** [reply]
Is of course better than nothing but mass/bulk enable/disable/delete workflows through UI would be much better.

<details><summary>I'm currently using this script to mass-delete workflow runs.</summary>

```bash
#!/usr/bin/env bash

# Delete workflow runs - dwr

# Given an "owner/repo" name, such as "qmacro/thinking-aloud",
# retrieve the workflow runs for that repo and present them in a
# list. Selected runs will be deleted. Uses the GitHub API.

# Requires gh (GitHub CLI) and jq (JSON processor)

# First version

set -o errexit
set -o pipefail

declare repo=${1:?No owner/repo specified}

jqscript() {

  #  ? : Error Suppression / Optional Operator : The ? operator, used as EXP?, is shorthand for try EXP.
  # // : Alternative operator: It's like ?? in PHP or JS.
  /bin/cat << EOF
      def symbol:
        sub(""; "")? // "NULL" |
        sub("skipped"; "SKIP") |
        sub("success"; "GOOD") |
        sub("failure"; "FAIL");

      def tz:
        gsub("[TZ]"; " ");

      def lf:
        gsub("\n.*"; "");

      .workflow_runs[]
        | [
            (.conclusion | symbol),
            (.status),
            (.created_at | tz),
            .name,
            (.head_commit.message | lf),
            .id
#            .event,
          ]
        | @tsv
EOF

}

selectruns() {

  gh api --method GET "/repos/$repo/actions/runs" -F page=1 -F per_page=90 \
    | jq -r -f <(jqscript) \
    | fzf --multi --reverse

}

deleterun() {

  local run id result
  run=$1
  id="$(cut -f 6 <<< "$run")"

  gh api -X DELETE "/repos/$repo/actions/runs/$id" --silent
  [[ $? = 0 ]] && result="OK!" || result="BAD"

  printf "%s\t%s\n" "$result" "$run"

}

deleteruns() {

  local id
  while read -r run; do
    deleterun "$run"
#    sleep 0.05
  done

}

main() {

  selectruns | deleteruns

}

main
```

</details>

But mass enable/disable workflows is a pain currently.

**Reply by jsoref** [reply]
💯
